collectors/nodestats/pipeline_subcollector.go

```diff
@@ -72,10 +76,35 @@ func (collector *PipelineSubcollector) Collect(pipeStats *responses.SinglePipeli
 		ch <- prometheus.NewMetricWithTimestamp(*pipeStats.Reloads.LastFailureTimestamp, prometheus.MustNewConstMetric(collector.ReloadsLastFailureTimestamp, prometheus.GaugeValue, 1, pipelineID))
 	}
 
+	if pipeStats.Reloads.LastSuccessTimestamp != nil {
+		ch <- prometheus.NewMetricWithTimestamp(*pipeStats.Reloads.LastSuccessTimestamp, prometheus.MustNewConstMetric(collector.ReloadsLastSuccessTimestamp, prometheus.GaugeValue, 1, pipelineID))
+	}
+	if pipeStats.Reloads.LastFailureTimestamp != nil {
+		ch <- prometheus.NewMetricWithTimestamp(*pipeStats.Reloads.LastFailureTimestamp, prometheus.MustNewConstMetric(collector.ReloadsLastFailureTimestamp, prometheus.GaugeValue, 1, pipelineID))
+	}
+
 	ch <- prometheus.MustNewConstMetric(collector.QueueEventsCount, prometheus.CounterValue, float64(pipeStats.Queue.EventsCount), pipelineID)
 	ch <- prometheus.MustNewConstMetric(collector.QueueEventsQueueSize, prometheus.CounterValue, float64(pipeStats.Queue.QueueSizeInBytes), pipelineID)
 	ch <- prometheus.MustNewConstMetric(collector.QueueMaxQueueSizeInBytes, prometheus.CounterValue, float64(pipeStats.Queue.MaxQueueSizeInBytes), pipelineID)
 
 	collectingEnd := time.Now()
 	log.Printf("collected pipeline stats for pipeline %s in %s", pipelineID, collectingEnd.Sub(collectingStart))
 }
+
+func (collector *PipelineSubcollector) isPipelineHealthy(pipeReloadStats responses.PipelineReloadResponse) float64 {
+	// 1. If last failure timestamp (or both) are nil, the pipeline is healthy
+	if pipeReloadStats.LastFailureTimestamp == nil {
+		return 1
+	// 2. If last_failure_timestamp is set and last success timestamp is nil, the pipeline is unhealthy
+	} else if pipeReloadStats.LastFailureTimestamp != nil && pipeReloadStats.LastSuccessTimestamp == nil {
+		return 0
+	// 3. If last_success_timestamp < last_failure_timestamp, the pipeline is unhealthy
+	} else if pipeReloadStats.LastSuccessTimestamp.Before(*pipeReloadStats.LastFailureTimestamp) {
+		return 0
+	// 4. If last_success_timestamp > last_failure_timestamp, the pipeline is healthy
+	} else if pipeReloadStats.LastSuccessTimestamp.After(*pipeReloadStats.LastFailureTimestamp) {
+		return 1
+	}
+	// Missing field, likely due to version incompatibility - lacking information, assume healthy
+	return 1
```

Comment:
This line, which is actually impossible now (due to line 96) is needed to avoid a "missing return" error. Yet [CodeCov dings it for lacking test coverage](https://app.codecov.io/gh/kuskoman/logstash-exporter/pull/94#e775519d92342df8508d8c9691ec0869-R109).  What's your recommendation? (I'm probably missing something obvious. Also, as case 1 is the normal path, I wanted to avoid checking all if/else conditions when not necessary.)

Reply:
@excalq I will handle that